torrent/torrent.go

```diff
@@ -432,6 +446,33 @@ func (t *torrent) FilePaths() ([]string, error) {
 	return filePaths, nil
 }
 
+type fileStat struct {
+	Path            string
+	BytesDownloaded int64
+	BytesTotal      int64
+}
+
+func (t *torrent) FileStats() ([]fileStat, error) {
```

Comment:
Can you change the API similar to the one you proposed in https://github.com/cenkalti/rain/issues/93#issuecomment-1471902176 ?

```
func (Torrent) Files() []File
func (File) Name() string
func (File) Stat() FileStat

type FileStat struct {
    BytesDownloaded int64
    BytesTotal int64
}
```

Reply:
Ok, no problem. I had wired things up this way as it allows querying for the completion data even if the torrent isn't in a running state. I had a followup change prepared to roundtrip the total bytes completed through the boltdb persistance layer as well. But it is simpler to only provide this when the torrent is running (incl seeding), so I'll start there. Ignore this pull request. I'll send a new one.